backends/qualcomm/passes/build_quant_io.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Qualcomm Innovation Center, Inc.
+# All rights reserved
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import torch
+
+from executorch.exir.pass_base import ExportPass, PassResult
+from executorch.exir.tensor import TensorSpec
+
+from .utils import q_io_key
+
+
+class BuildQuantIo(ExportPass):
+    """
+    To make lowering process correct, the pass assign the correct quantized dtype to spec of call_delegate.
+    """
+
+    def __init__(self):
+        super(BuildQuantIo, self).__init__()
+
+    def _make_spec(self, x):
+        if isinstance(x, torch.Tensor):
+            return TensorSpec.from_tensor(x)
+        elif isinstance(x, (int, bool, float)):
+            return x
+        else:
+            return None
+
+    def _build(self, graph_module: torch.fx.GraphModule) -> torch.fx.GraphModule:
+        # forcely update delegate node's meta['spec'] to get correct output
+        # tensor size in runtime
+        call_delegate = [
+            node
+            for node in graph_module.graph.nodes
+            if node.op == "call_function" and node.name == "executorch_call_delegate"
```

Comment:
Did you expect to run this pass after to_backend? Was it because of some issues you run into?

Reply:
Thank you for pointing it out.
Yes, we intentionally invoke this pass after to_backend to change the io data type to non-fp type.

Reply:
I'm trying to understand the idea. It seems like foldqdq will remove all the q/dq nodes, and insert io qdq passes will insert q/dq nodes for the subgraph inside qnn backend again. In this case, why would the io data type become non-fp type?

Reply:
Right, the flow you mentioned above is exactly what we have done before. Yet in LLAMA we specifically [deal with its KV IOs](https://github.com/pytorch/executorch/pull/4142/files#diff-716ed91cb062d062c9434ce96b213e0941cc24e58314de3686acfce5f53c4bb0R251). Quantizing and dequantizing are meaningless for these IOs. Therefore we keep them as quantized type all the time. 

Here is what we have done for these kind of tensors.
1.Tag IO nodes [as a quantized type](https://github.com/pytorch/executorch/pull/4142/files#diff-716ed91cb062d062c9434ce96b213e0941cc24e58314de3686acfce5f53c4bb0R340) in examples/qualcomm/llama2/llama.py:340
2. Skip insert quantized/dequantized nodes in [insert_io_qdq](https://github.com/pytorch/executorch/pull/4142/files#diff-482726e32a75c5a6f3c6439ea06a97383d6865237e8ed8a5d6193cb43b268f2dR115)
3. Set correct quantized type to [their spec](https://github.com/pytorch/executorch/pull/4142/files#diff-716ed91cb062d062c9434ce96b213e0941cc24e58314de3686acfce5f53c4bb0R316) in examples/qualcomm/llama2/llama.py:316

Feel free to let me know if anything is unclear to you. Thank you. :D